### PR TITLE
OpenJDK with crac enabled

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [ ] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ ] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

#### For package updates (renames) in the base images

When updating packages part of base images (i.e. cgr.dev/chainguard/wolfi-base or ghcr.io/wolfi-dev/sdk)
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk images successfully build
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk contain no obsolete (no longer built) packages
- [ ] Upon launch, does `apk upgrade --latest` successfully upgrades packages or performs no actions

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: openjdk-crac-17.yaml [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+      - zip
+      - crac-criu
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/openjdk/crac.git
+      tag: jdk-${{vars.mangled-package-version}}
+      expected-commit: 6765f902505fbdd02f25b599f942437cd805cad1
+
```
Comment (xnox) [inline openjdk-crac-17.yaml:54]: I expect patched code to be build here. I.e. crac-17 branch or crac branch. As the tags seem to point at stock vanilla JDK.

**Review by xnox — CHANGES_REQUESTED** [review]

**Comment by xnox** [comment]
Note given it is a patchset we need to manage lifecycle of both openjdk and the crac patchset. As in have them separated somehow and apply / rebase as new openjdk get released.

File: openjdk-crac-17.yaml [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+      - libxext-dev
+      - libxi-dev
+      - libxrandr-dev
+      - libxrender-dev
+      - libxt-dev
+      - libxtst-dev
+      - openjdk-16
+      - openjdk-16-default-jvm
+      - zip
+      - crac-criu
```
Comment (EyeCantCU) [inline openjdk-crac-17.yaml:46]: Let's remove this at buildtime and provide at runtime, create a symlink to jdk/lib/criu

**Comment by tuananh** [comment]
subscribing 

Related issue: https://github.com/wolfi-dev/os/issues/27342

**Pull request closed without merging**